### ISSUE TITLE
systray invisible

### ISSUE THREAD

**Issue by foolishgrunt** [lxde/lxqt#696]
When I add the 'System tray' (systray) widget to LXQT-Panel, nothing appears. I know it's there, because of the two screenshots I've attached.

First, the 'Add Panel Widgets' dialog shows an active instance, and if I attempt to add more widgets they will all be represented in the dialog.

Second, I know it's not exactly easy to make out, but there's a small gap present between the edge of my 'Task manager' widget and my 'Volume control' widget. That represents where the systray widget is supposed to be. If I right click on that space, I get a 'System tray' popup dialog with options to move or remove the widget.

So it's definitely there. It's just useless. If I open any program with a systray icon (e.g. Clementine, caffeine-ng) the icon does not appear in the systray. No amount of moving the widget helps, and neither does resizing the panel. Does this problem sound familiar to anyone, or can anyone help me identify a solution?

I'm running Arch Linux x86_64, using LXQT (and LXQT-Panel) version 0.9
![screenshot](https://cloud.githubusercontent.com/assets/1423934/8200107/e2322f38-1474-11e5-810e-323022ca18ab.png)
![useless](https://cloud.githubusercontent.com/assets/1423934/8200106/e22f8c9c-1474-11e5-820b-d4b7f4053f7f.png)

**Comment by palinek** [comment]
@foolishgrunt have you only one panel with one instance of 'System tray' plugin running?

**Comment by pmattern** [comment]
Is this by any chance in a KVM/QEMU VM featuring the QXL virtual graphics adapter and its corresponding driver? The latter doesn't display the icons in LXQt's System tray plugin exactly as you describe.
To verify you could simply remove its package, most likely xf86-video-qxl[-git] on Arch Linux. The X server will use the modesetting driver (/usr/lib/xorg/modules/drivers/modesetting_drv.so) from package xorg-server (used to be in xf86-video-modesetting) which doesn't suffer from this problem.

**Comment by foolishgrunt** [comment]
@palinek I currently have two panels (one for each monitor I'm using), with one instance of System tray per panel. However, I experienced the same issue when I was only using one panel.

@pmattern No, this is not a virtual machine, and I do not have the QXL driver installed.

**Comment by paulolieuthier** [comment]
@foolishgrunt it's impossible to have more than one systray plugin active. That's true in any desktop environment. This is already fixed in the development branch, you'll get updates soon, hopefully.

**Comment by foolishgrunt** [comment]
@paulolieuthier: Thanks for the heads up, I'll be on the lookout for that.
However, as I mentioned in my last post, the invisibility problem exists even with only one panel and only one systray instance. I removed the second panel and systray after reading your last post, and have confirmed that I still cannot see anything in the systray. Unless I somehow broke something *permanently* by adding the second panel and systray instance, those problems do not appear to be related.

**Comment by paulolieuthier** [comment]
Oh, sorry. Reopening then.

> Unless I somehow broke something permanently by adding the second panel and systray instance, those problems do not appear to be related.

I don't that can happen. You are using LXQt packages from the official repos, right? I don't remember of any issue like that one, even before 0.9. The best thing you can do is to install the AUR packages to see if the problem persists.

**Comment by pmattern** [comment]
I think another thing worth trying is to delete ~/.config/lxqt so as to see whether there's some rotten remnant from the times when two panels were in use.
Should be done while no session is running.
An alternative would be to create a new user and see whether the problem exists in his sessions as well.

**Comment by foolishgrunt** [comment]
To answer @paulolieuthier question, yes I'm using the official repo packages. I was going to uninstall them and try the development packages from AUR, but thought I would try @pmattern suggestion first. I switched to a different TTY and deleted ~/.config/lxqt, and when I went back to TTY and signed in, my systray was instantly visible and usable.

So thanks for your help guys! I guess I did accidentally create some rotten remnants.
